presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TableScanMatcher.java

```diff
@@ -39,9 +50,11 @@ public boolean matches(PlanNode node, Session session, Metadata metadata, Symbol
             TableScanNode tableScanNode = (TableScanNode) node;
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableScanNode.getTable());
             String actualTableName = tableMetadata.getTable().getTableName();
-            if (expectedTableName.equalsIgnoreCase(actualTableName)) {
-                return true;
+            boolean matches = expectedTableName.equalsIgnoreCase(actualTableName);
+            if (matches && predicate.isPresent() && tableScanNode.getOriginalConstraint() != null) {
```

Comment:
I wanted to refrain from validating tuple domains as I thought it might be to complicated for this kind of tests. I fixed that, now `currentConstraint` is validated.

Currently in `TestLogicalPlanner` the `localQueryRunner` is used, which uses `tpch` connector, and so it is should be safe to use that for testing.